new plugin: betterBlockedContext

sillyElv opened the pull request:
This plguin improves discord's new blocked and ignored view by allowing you to access the profile and DMs through the menu. This was possible before Discord moved the Blocked Users section.

sillyElvy committed:
new plugin: betterBlockedContext

sillyElvy committed:
few improvements to the plugin

sadan4 commented on src/plugins/betterBlockedContext/index.ts:
```diff
@@ -0,0 +1,84 @@
+        {
+            find: ',["user"])',
+            replacement: {
+                match: /(?<=\i,\i]=\i.useState\().*?]\);/,
+                replace: "false);"
```
i feel like this should anchor to something that isn't just useState, as it could match any newly added useState calls and error

sillyElvy committed:
better anchor for match

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
:wires:

InSimpleTermsJordan commented:
absolutely necessary

sillyElvy committed:
add a button to access DMs faster

sillyElvy committed:
add a button to access DMs faster

sillyElvy committed:
add a warning for unblocking/unignoring through the list.

sillyElvy committed:
add a setting for general warning on unblocking

sillyElvy committed:
fix description, fix naming

sillyElvy committed:
shorten regex a bit

EepyElvyra commented:
this should be ready now, i love adding useless shit

sillyElvy committed:
add shift-click to everything

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Improve comment

sillyElvy committed:
Fix comment again

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
minor improvements i think

sillyElvy committed:
group patch

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
fix patches for latest discord updates

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Improve description

sillyElvy committed:
Better desc agane

sillyElvy committed:
Remove outdated comment

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

EepyElvyra commented:
@CodiumAI-Agent /review

CodiumAI-Agent commented:
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 4 🔵🔵🔵🔵⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<details><summary><a href='https://github.com/Vendicated/Vencord/pull/3346/files#diff-4bb712acb8940a8db2bce1992b6e950603850326a311547dea686d91b19b45cdR65-R78'><strong>Fragile Regex</strong></a>

Regex-based patching is brittle and may break with future Discord updates. Consider using AST-based patching or more resilient selectors instead of fragile lookbehind patterns.
</summary>

```typescript
{
    find: ".lastRow]",
    replacement: [
        // Allow the user profile to be opened from the blocklist.
        {
            match: /(?<=className:\i.userInfo,)(?=children:.{0,20}user:(\i))/,
            replace: "style:{cursor:'pointer'},onClick:() => $self.openUserProfile($1),"
        },
        // Add an extra message button into the blocklist for easy access to past DMs, in case those are needed.
        {
            match: /(?<=children:null!=(\i).globalName\?.+?}\),).*?(\{color:.{0,65}?string\((\i).+?"8wXU9P"]\)})\)/,
            replace: "$self.generateButtons({user:$1, originalProps:$2, isBlocked:$3})",
        }
    ],
```

</details>

<details><summary><a href='https://github.com/Vendicated/Vencord/pull/3346/files#diff-4bb712acb8940a8db2bce1992b6e950603850326a311547dea686d91b19b45cdR155-R162'><strong>Props Mutation</strong></a>

generateButtons directly mutates `originalProps`, which can introduce unintended side effects. Clone or spread props before modifying.
</summary>

```typescript
const { user, originalProps, isBlocked } = props;

if (settings.store.unblockButtonDanger) originalProps.color = Button.Colors.RED;

if (settings.store.showUnblockConfirmation || settings.store.showUnblockConfirmationEverywhere) {
    const originalOnClick = originalProps.onClick!;
    originalProps.onClick = e => {
        if (!isBlocked) return originalOnClick(e);
```

</details>

<details><summary><a href='https://github.com/Vendicated/Vencord/pull/3346/files#diff-4bb712acb8940a8db2bce1992b6e950603850326a311547dea686d91b19b45cdR192-R199'><strong>User Null Safety</strong></a>

openConfirmationModal assumes `user` and `user.username` are always defined. Add null checks or default values to avoid runtime errors.
</summary>

```typescript
openConfirmationModal(event: MouseEvent, callback: () => any, user: User|string, isSettingsOrigin: boolean = false) {
    if (event.shiftKey && settings.store.allowShiftUnblock) return callback();

    if (typeof user === "string") {
        user = UserStore.getUser(user);
    }

    return openModal(m => <ConfirmationModal
```

</details>

</td></tr>
</table>

Vendicated commented:
hi! please do not use AI for writing pull requests. this pull request is **__closed__**!

CodiumAI-Agent commented:
**[Persistent review](https://github.com/Vendicated/Vencord/pull/3346#issuecomment-2869003228)** updated to latest commit https://github.com/Vendicated/Vencord/commit/dc70d5e5c7b1a830f69a897ec70277c9bb2bc743

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

EepyElvyra commented:
> hi! please do not use AI for writing pull requests. this pull request is **__closed__**!

The fuck you mean, I don't even code with ai 😭

InSimpleTermsJordan commented:
> hi! please do not use AI for writing pull requests. this pull request is **closed**!

This bot doesn't seem to be actually doing any writing? it seems to just be reviewing, not actually changing anything

Vendicated commented:
https://github.com/user-attachments/assets/6fe89627-ab00-42b9-861d-cf2d1e2c87f3

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
Fix my retardation

sillyElvy committed:
fix patch; fix wordings

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
(slightly) saner patch

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
I was enlightened

Nuckyz committed:
Fix IgnoreActivities

thororen1234 committed:
fix failing to find the Select component (#3631)

sillyElvy committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
fix injection of buttons; profile fix pending

Vendicated committed:
Merge branch 'dev' into feat/blockedContext

sillyElvy committed:
add in a regex warcrime cuz funni

gonna remove next comiet

sillyElvy committed:
remove feature as its not cleanly possible anymore

users will have to rely on other options to unblock/read DMs but this plugin provides those

EepyElvyra commented:
This one should be ready and adapted to the latest changes again if I'm not entirely stupid

sillyElvy committed:
remove stale code
